redaxo/src/addons/media_manager/lib/managed_media.php

```diff
@@ -165,18 +165,37 @@ public function setFormat($format)
 
     public function sendMedia($sourceCacheFilename, $headerCacheFilename, $save = false)
     {
-        $src = $this->getSource();
-
-        $this->prepareHeaders($src);
+        if ($this->asImage) {
+            $src = $this->getSource();
+            $this->prepareHeaders($src);
+        } else {
+            // Content-Length will be overwritten by rex_response::sendFile
+            $this->prepareHeaders('');
+        }
 
         rex_response::cleanOutputBuffers();
         foreach ($this->header as $t => $c) {
-            header($t . ': ' . $c);
+            rex_response::setHeader($t, $c);
         }
-        echo $src;
 
-        if ($save) {
-            $this->saveFiles($src, $sourceCacheFilename, $headerCacheFilename);
+        if ($this->asImage) {
```

Comment:
Warum brauchen wir das if hier?
Können wir nicht rex_response->sendContent() verwenden und dort ebenfalls ein range request unterstützen?

Reply:
stimmt, das macht mehr sinn 👍

Reply:
Ich hatte dies an der Stelle getrennt, da ich sonst ein MemoryProblem bekomme, wenn ein komplettes Video >150MB in den RAM geladen wird.
siehe https://github.com/redaxo/redaxo/issues/2267

Reply:
durch die trennung mittels `if ($this->asImage)` kann ich ermitteln, ob die Datei über den MediaManager bearbeitet wurde (Bild-Effekte, etc.). Anhand dessen bin ich davon ausgegangen, da Videos keine Bildeffekte bekommen, dass ich diese direkt von der OrgFile aus streamen kann.

Reply:
> durch die trennung mittels `if ($this->asImage)` kann ich ermitteln, ob die Datei über den MediaManager bearbeitet wurde (Bild-Effekte, etc.). Anhand dessen bin ich davon ausgegangen, da Videos keine Bildeffekte bekommen, dass ich diese direkt von der OrgFile aus streamen kann.

ob diese annahme passt, kann ich nicht beurteilen. @gharlan  vllt?

Reply:
ich bin jetzt für diesen PR davon ausgegangen, dass das MemoryProblem nicht auftritt. Das sollte man in einem getrennten PR lösen.